diff --git a/template/magisk_module/customize.sh b/template/magisk_module/customize.sh
index 53ded41..0527259 100644
--- a/template/magisk_module/customize.sh
+++ b/template/magisk_module/customize.sh
@@ -78,24 +78,24 @@ if [ "$IS64BIT" = true ]; then
   [ "$FLAVOR" = "riru" ] && mv "$LIB64_DEST/$LIB64_NAME" "${LIB64_DEST}/lib${MODULE_ID}.so"
 fi
 
-ui_print "- Extracting bundled frida gadget"
+ui_print "- Extracting bundled gadget"
 
 mkdir -p "$TMP_MODULE_DIR"
-extract "$ZIPFILE" "gadget/libgadget-$ARCH.so.xz" "$TMP_MODULE_DIR" true
-mv "$TMP_MODULE_DIR/libgadget-$ARCH.so.xz" "$TMP_MODULE_DIR/libgadget.so.xz"
-rm "$TMP_MODULE_DIR/libgadget.so"
-$BUSYBOX_BIN unxz "$TMP_MODULE_DIR/libgadget.so.xz"
-rm "$TMP_MODULE_DIR/libgadget.so.xz"
+extract "$ZIPFILE" "gadget/lib__BRAND__gadget-$ARCH.so.xz" "$TMP_MODULE_DIR" true
+mv "$TMP_MODULE_DIR/lib__BRAND__gadget-$ARCH.so.xz" "$TMP_MODULE_DIR/lib__BRAND__gadget.so.xz"
+rm -f "$TMP_MODULE_DIR/lib__BRAND__gadget.so"
+$BUSYBOX_BIN unxz "$TMP_MODULE_DIR/lib__BRAND__gadget.so.xz"
+rm -f "$TMP_MODULE_DIR/lib__BRAND__gadget.so.xz"
 
 if [ "$IS64BIT" = true ]; then
   ARCH32="arm"
   [ "$ARCH" = "x64" ] && ARCH32="x86"
 
-  extract "$ZIPFILE" "gadget/libgadget-$ARCH32.so.xz" "$TMP_MODULE_DIR" true
-  mv "$TMP_MODULE_DIR/libgadget-$ARCH32.so.xz" "$TMP_MODULE_DIR/libgadget32.so.xz"
-  rm "$TMP_MODULE_DIR/libgadget32.so"
-  $BUSYBOX_BIN unxz "$TMP_MODULE_DIR/libgadget32.so.xz"
-  rm "$TMP_MODULE_DIR/libgadget32.so.xz"
+  extract "$ZIPFILE" "gadget/lib__BRAND__gadget-$ARCH32.so.xz" "$TMP_MODULE_DIR" true
+  mv "$TMP_MODULE_DIR/lib__BRAND__gadget-$ARCH32.so.xz" "$TMP_MODULE_DIR/lib__BRAND__gadget32.so.xz"
+  rm -f "$TMP_MODULE_DIR/lib__BRAND__gadget32.so"
+  $BUSYBOX_BIN unxz "$TMP_MODULE_DIR/lib__BRAND__gadget32.so.xz"
+  rm -f "$TMP_MODULE_DIR/lib__BRAND__gadget32.so.xz"
 fi
 
 extract "$ZIPFILE" "config.json.example" "$TMP_MODULE_DIR" true
