diff --git a/src/linux/helpers/zymbiote.c b/src/linux/helpers/zymbiote.c
index b9fa5268..485a28a9 100644
--- a/src/linux/helpers/zymbiote.c
+++ b/src/linux/helpers/zymbiote.c
@@ -37,7 +37,7 @@ struct _ZymbioteContext
 
 ZymbioteContext zymbiote =
 {
-  .socket_path = "/frida-zymbiote-00000000000000000000000000000000",
+  .socket_path = "/__PLACEHOLDER_ZYMBIOTE__-00000000000000000000000000000000",
 };
 
 int frida_zymbiote_replacement_setargv0 (JNIEnv * env, jobject clazz, jstring name);
diff --git a/src/linux/linux-host-session.vala b/src/linux/linux-host-session.vala
index 44bd9400..09bf96d1 100644
--- a/src/linux/linux-host-session.vala
+++ b/src/linux/linux-host-session.vala
@@ -64,12 +64,12 @@ namespace Frida {
 			var blob64 = Frida.Data.Agent.get_frida_agent_64_so_blob ();
 			var emulated_arm = Frida.Data.Agent.get_frida_agent_arm_so_blob ();
 			var emulated_arm64 = Frida.Data.Agent.get_frida_agent_arm64_so_blob ();
-			agent = new AgentDescriptor (PathTemplate ("frida-agent-<arch>.so"),
+			agent = new AgentDescriptor (PathTemplate ("dyegt-agent-<arch>.so"),
 				new Bytes.static (blob32.data),
 				new Bytes.static (blob64.data),
 				new AgentResource[] {
-					new AgentResource ("frida-agent-arm.so", new Bytes.static (emulated_arm.data), tempdir),
-					new AgentResource ("frida-agent-arm64.so", new Bytes.static (emulated_arm64.data), tempdir),
+					new AgentResource ("dyegt-agent-arm.so", new Bytes.static (emulated_arm.data), tempdir),
+					new AgentResource ("dyegt-agent-arm64.so", new Bytes.static (emulated_arm64.data), tempdir),
 				},
 				AgentMode.INSTANCED,
 				tempdir);
@@ -412,10 +412,10 @@ namespace Frida {
 			unowned string name;
 			switch (cpu_type_from_pid (pid)) {
 				case Gum.CpuType.IA32:
-					name = "frida-agent-arm.so";
+					name = "dyegt-agent-arm.so";
 					break;
 				case Gum.CpuType.AMD64:
-					name = "frida-agent-arm64.so";
+					name = "dyegt-agent-arm64.so";
 					break;
 				default:
 					throw new Error.NOT_SUPPORTED ("Emulated realm is not supported on this architecture");
@@ -1528,7 +1528,7 @@ namespace Frida {
 			ensure_request = new Promise<bool> ();
 
 			if (server_name == null) {
-				string name = "/frida-zymbiote-" + Uuid.string_random ().replace ("-", "");
+				string name = "/__PLACEHOLDER_ZYMBIOTE__-" + Uuid.string_random ().replace ("-", "");
 				var address = new UnixSocketAddress.with_type (name, -1, UnixSocketAddressType.ABSTRACT);
 
 				try {
@@ -1954,7 +1954,7 @@ namespace Frida {
 
 			unowned uint8[] payload_template = blob.data[text.file_offset:text.file_offset + text.file_size];
 
-			void * p = memmem (payload_template, "/frida-zymbiote-00000000000000000000000000000000".data);
+			void * p = memmem (payload_template, "/__PLACEHOLDER_ZYMBIOTE__-00000000000000000000000000000000".data);
 			assert (p != null);
 			size_t data_offset = (uint8 *) p - (uint8 *) payload_template;
 
